src/manageiq_client/api.py

```diff
@@ -457,11 +482,18 @@ def __init__(self, obj):
 
     def reload(self):
         self._obj.reload_if_needed()
+        reloaded_actions = []
         for action in self._obj._actions:
-            setattr(
-                self,
-                action["name"],
-                Action(self, action["name"], action["method"], action["href"]))
+            # There can be multiple actions with the same name and different methods
+            # (e.g. actions "delete" with method POST and DELETE).
+            # This makes sure that the attribute refers to the first action and is not redefined
+            # by other action with the same name and different method.
```

Comment:
This just picks first instead of last, but it's still random?
Could define some fixed preference order on methods, but still the chosen method might shift when API starts supporting more methods for same action...

Does this whole method-agnostic exercise really work — can we provide params in same format to different http methods?  Does it buy us much?
Is `foos.action.bar(a=1, b=2)` better than explicit `foos.post(action="bar", resource=dict(a=1, b=2))`?
[The latter doesn't work now, need `api.post(href=foos._href, ...)` or something, but is doable.]